inducer/meshmode
# Tensor Product Additions

Implements a couple of small things for tensor products:

* Differentiation using inducer/modepy#13 (may need some version bumps?)
* Some version of `face_vertex_indices` that seems to allow building up the facial adjacency
* Expanded the check in `_test_node_vertex_consistency`
* Expanded some tests to check this stuff
* (inducer in alexfikl/meshmode#1) Update `make_face_restriction`.
* Unloaded all "reference element" details to `modepy` (needs inducer/modepy#16).

Some remaining things to do:

* [x] Refinement doesn't support tensor product elements.
    - WIP at alexfikl/meshmode#3
* [x] Use inducer/modepy#16 shapes and function spaces in more places.
* [x] Test grudge with new versions of meshmode and modepy.
    - See inducer/grudge#35
* [x] Test pytential with new versions of meshmode and modepy.
    - See inducer/pytential#44
